src/cpp/RiderLink/Source/RiderGameControlExtension/Private/RiderGameControlExtension.cpp

```diff
@@ -310,107 +312,80 @@ void FRiderGameControlExtensionModule::StartupModule()
     }
 
     const rd::Lifetime NestedLifetime = FRiderLinkModule::Get().CreateNestedLifetime();
-    FRiderLinkModule::Get().RdConnection.UnrealToBackendModel.get_play().advise(
-        NestedLifetime, [this](int playValue)
+    FRiderLinkModule::Get().RdConnection.UnrealToBackendModel.get_playStateFromRider().advise(
+        NestedLifetime, [this](Jetbrains::EditorPlugin::PlayState State)
         {
-            if (PlayFromUnreal)
-                return;
-            FSetForTheScope s(PlayFromRider);
-
-            if (!playValue && GUnrealEd && GUnrealEd->PlayWorld)
-            {
-                GUnrealEd->RequestEndPlayMap();
-            }
-            else if (playValue == 1 && GUnrealEd)
+            if(!GUnrealEd) return;
+            switch(State)
             {
+            case Jetbrains::EditorPlugin::PlayState::Idle:
+                if (GUnrealEd->PlayWorld)
+                {
+                    GUnrealEd->RequestEndPlayMap();
+                }
+                break;
+            case Jetbrains::EditorPlugin::PlayState::Play:
                 if (GUnrealEd->PlayWorld &&
-                    GUnrealEd->PlayWorld->bDebugPauseExecution)
+                    GUnrealEd->PlayWorld->IsPaused())
                 {
                     GUnrealEd->PlayWorld->bDebugPauseExecution = false;
+                    // Simply switching flag doesn't work, `ResumePIE` delegate won't be triggered
+                    GUnrealEd->PlaySessionResumed();
                 }
                 else
                 {
                     const int Mode = FRiderLinkModule::Get()
                                      .RdConnection.UnrealToBackendModel.get_playMode().get();
                     RequestPlay(Mode);
                 }
-            }
-            else if (playValue == 2 && GUnrealEd && GUnrealEd->PlayWorld)
-            {
-                GUnrealEd->PlayWorld->bDebugPauseExecution = true;
-            }
-        });
-
-    FRiderLinkModule::Get().RdConnection.UnrealToBackendModel.get_play().advise(
```

Comment:
Was it written twice?

Reply:
Yeah, looks like I've screwed up when I was dividing RiderLink into separate modules